book/src/leader-validator-transition.md

```diff
@@ -0,0 +1,157 @@
+# Leader and Validator Transition
+
+A fullnode operates in two modes, leader and validator.  The modes overlap in
+code and in function, but have different behaviors.  The goal for this design is
+to allow the two modes to transition between states while reusing the computed
+bank state.
+
+
+## Validator
+
+A validator operates on may different concurrent forks of the bank state until
+it can prove a PoH record at a height that is the scheduled slot to be a leader.
+
+## Leader
+
+A leader operates on a single fork for a specific PoH slot.
+
+
+## PoH
+
+Two different objects for managing PoH for leaders and validators.  Both only
+work for a specific range of ticks and will error out and send an exit signal
```

Comment:
"send an exit signal" is an implementation detail. Not sure that helps clarify anything here.

Reply:
@garious This design is intentionally a bit implementation heavy.  The control for the loop is designed around a single thread that loops and transitions between states.